src/Azure/Orleans.Streaming.AzureStorage/Providers/Streams/AzureQueue/AzureQueueStreamProviderUtils.cs

```diff
@@ -17,52 +17,44 @@ public class AzureQueueStreamProviderUtils
         /// Helper method for testing. Deletes all the queues used by the specifed stream provider.
         /// </summary>
         /// <param name="loggerFactory">logger factory to use</param>
-        /// <param name="providerName">The Azure Queue stream privider name.</param>
-        /// <param name="deploymentId">The deployment ID hosting the stream provider.</param>
+        /// <param name="queueNamePrefix">The Azure Queue stream queue name prefix.</param>
         /// <param name="storageConnectionString">The azure storage connection string.</param>
-        public static async Task DeleteAllUsedAzureQueues(ILoggerFactory loggerFactory, string providerName, string deploymentId, string storageConnectionString)
+        public static async Task DeleteAllUsedAzureQueues(ILoggerFactory loggerFactory, string queueNamePrefix, string storageConnectionString)
         {
-            if (deploymentId != null)
-            {
-                // TODO: Do not assume defaults !? - jbragg
-                var queueMapper = new HashRingBasedStreamQueueMapper(new HashRingStreamQueueMapperOptions(), providerName);
-                List<QueueId> allQueues = queueMapper.GetAllQueues().ToList();
-
-                var deleteTasks = new List<Task>();
-                foreach (var queueId in allQueues)
-                {
-                    var manager = new AzureQueueDataManager(loggerFactory, queueId.ToString(), deploymentId, storageConnectionString);
-                    deleteTasks.Add(manager.DeleteQueue());
-                }
+            // TODO: Do not assume defaults !? - jbragg
+            var queueMapper = new HashRingBasedStreamQueueMapper(new HashRingStreamQueueMapperOptions(), queueNamePrefix);
+            List<QueueId> allQueues = queueMapper.GetAllQueues().ToList();
 
-                await Task.WhenAll(deleteTasks);
+            var deleteTasks = new List<Task>();
+            foreach (var queueId in allQueues)
+            {
+                var manager = new AzureQueueDataManager(loggerFactory, queueId.ToString(), storageConnectionString);
```

Comment:
The queue name needs to map to a queueId but does not need to be the name of the queueId.  QueueId is unnecessarily constrained.  A team with an existing set of queues "A", "B", "C", which they wanted to process using Orleans streams would not be supported by queue names constrained by QueueId naming.

Reply:
I'm not sure what feature you are suggesting here. If a team want to configure azure queue name completely, then they need to give a list of their queue name to AzureQueueStreamProvider which has the same queue amount of the queues configured with AzureQueueStreamProvider. This requires user to have certain level of understanding for AzureQueueStreamProvider. 

So let's say there's a new user just want to configure a AzureQueueStreamProvider with connection string and play with it after, how do we do in this situation? 

Just btw you might missed the context of this PR. The goal for this PR is to fix #4577 , which is to not use unsupported symbol in Azure queue name , while the old queue name is serviceId-providerName by default, so configuring azure queue name is hard. So serviceId and providerName should contain unsupported symbol. This is hard to know from the user's side since it is implementation details. So I think this PR solve that problem